packages/payment-processor/src/payment/erc20-hinkal-wallet.ts

```
@@ -0,0 +1,57 @@
+import { RelayerTransaction } from '@hinkal/client/dist/types/relay';
+
+export async function payErc20RequestHinkalWallet(
+  request: ClientTypes.IRequestData,
+  signerOrProvider: providers.Provider | Signer = getProvider(),
+  amount: BigNumberish,
+): Promise<providers.TransactionResponse> {
+  const signer = getSigner(signerOrProvider);
+
+  const { transactionHash } = await constructAndSendTransferOp(request, signer, amount);
+
+  return await signer.provider!.getTransaction(transactionHash);
```

Biome

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)